### PR TITLE
Readme

### PR DESCRIPTION
**Pull request by codeaye**

### CODE REVIEW

File: projects/mathjax-lib/README.md [hunk 1]
```diff
@@ -1,24 +1,144 @@
...
-Run `ng test mathjax-lib` to execute the unit tests via [Karma](https://karma-runner.github.io).
+### Usage
+When importing in the **root** module, the module should be configured
+with *.forRoot* method.
 
-## Further help
+```typescript
+import {NgModule} from '@angular/core';
+import {AppComponent} from './src/app/app.component';
+import {MathJaxModule} from './src/app/math-jax/math-jax.module';
```
Comment (sajivkumar) [inline projects/mathjax-lib/README.md]: replace with npm package 
`import { MathjaxModule } from 'mathjax-angular';`

File: projects/mathjax-lib/README.md [hunk 2]
```diff
@@ -1,24 +1,144 @@
-# MathjaxLib
+# mathjax-angular 
+This plugins implements the browser version of [Mathjax v3][1] into [Angular][2]
+## Feature
 
-This library was generated with [Angular CLI](https://github.com/angular/angular-cli) version 12.1.1.
+- Dynamically load MathJax library to your web application.
```
Comment (sajivkumar) [inline projects/mathjax-lib/README.md]: rephrase this to our own words

Comment (sajivkumar) [reply projects/mathjax-lib/README.md]: any web application
mention that as this lib is developed in angular12, this may not support old ie12 or later version.

File: projects/mathjax-lib/README.md [hunk 3]
```diff
@@ -1,24 +1,144 @@
...
+```
+yarn add mathjax-angular
+```
 
-Run `ng build mathjax-lib` to build the project. The build artifacts will be stored in the `dist/` directory.
+## Configure the module
 
-## Publishing
+Load the module in the `@NgModule` class of the application. You need
+to pass a `ModuleConfiguration` instance to the `config` method to
```
Comment (sajivkumar) [inline projects/mathjax-lib/README.md]: replace 
ModuleConfiguration with RootMathjaxConfig

File: projects/mathjax-lib/README.md [hunk 4]
```diff
@@ -1,24 +1,144 @@
...
+configure the module.
 
-After building your library with `ng build mathjax-lib`, go to the dist folder `cd dist/mathjax-lib` and run `npm publish`.
+### Example
 
-## Running unit tests
+We provide a premade exaple with the library itself, You can see the example [here][3], or alternatively run it locally by cloning the [repository][4] and running:
+```
+npm install&&npm run dev
```
Comment (sajivkumar) [inline projects/mathjax-lib/README.md]: change this to two steps
1) npm install - this is for first time only to make sure all the packages are to the latest version
2)npm run dev - will launch the application in development mode and opens in browser
or
2) npm run dev production - will launch application in production mode in your default browser

File: projects/mathjax-lib/README.md [hunk 5]
```diff
@@ -1,24 +1,144 @@
...
+```
 
-Run `ng test mathjax-lib` to execute the unit tests via [Karma](https://karma-runner.github.io).
+### Usage
+When importing in the **root** module, the module should be configured
+with *.forRoot* method.
```
Comment (sajivkumar) [inline projects/mathjax-lib/README.md:46]: you can optionally pass config. for this refer to majax site 
http://docs.mathjax.org/en/latest/web/configuration.html

current defaults to below
`{
  config: {
    loader: {
      load: ['output/svg', '[tex]/require', '[tex]/ams'],
    },
    tex: {
      inlineMath: [['$', '$']],
      packages: ['base', 'require', 'ams'],
    },
    svg: { fontCache: 'global' },
  },
  src: 'https://cdn.jsdelivr.net/npm/mathjax@3.0.0/es5/startup.js',
}`

Comment (sajivkumar) [reply projects/mathjax-lib/README.md:46]: "config" : refer - http://docs.mathjax.org/en/latest/web/configuration.html#configuring-and-loading-mathjax
'src' = source cdn

File: projects/mathjax-lib/README.md [hunk 6]
```diff
@@ -1,24 +1,144 @@
...
+})
+export class AppModule { }
+```
+
+When importing in a **child** module, the module must be configured to
+re-use the same module instance as the root module. So simply
+configure the module with the *.forChild* method.
+
+```typescript
+import {MathJaxModule} from './src/app/math-jax/math-jax.module';
```
Comment (sajivkumar) [inline projects/mathjax-lib/README.md:91]: `import { MathjaxModule } from 'mathjax-angular';`

File: projects/mathjax-lib/README.md [hunk 7]
```diff
@@ -1,24 +1,144 @@
...
+
+## Typeset an element
+
+Add the `mathjax` directive to elements which you want to apply
+MathJax typesetting on load.
+
+```html
+<div>normal text</div>
+
+<div mathjax>mathjax typesetting
```
Comment (sajivkumar) [inline projects/mathjax-lib/README.md]: remove this
in our implementation we need to pass mathjax value as attribute

Comment (sajivkumar) [reply projects/mathjax-lib/README.md]: `<div [mathjax]="content"></div>
// in your ts component
content = "$x = {-b \\pm \\sqrt{b^2-4ac} \\over 2a}$"
`

File: projects/mathjax-lib/README.md [hunk 8]
```diff
@@ -1,24 +1,144 @@
...
+$$
+x = 1
+$$
+
+\( y = 2 \)
+</div>
+```
+
+## Typesetting using expression
```
Comment (sajivkumar) [inline projects/mathjax-lib/README.md]: remove this. this is not supported

File: projects/mathjax-lib/README.md [hunk 9]
```diff
@@ -1,24 +1,144 @@
...
+MathJax Expression 2: \( {{ '{}' }} \)
+</div>
+```
+
+Insert the `{}` to the place you want, then surround it with a pair of
+MathJax delimiter.
+
+You need to escape it in Angular template.
+
+## Manually trigger MathJax typesetting
```
Comment (sajivkumar) [inline projects/mathjax-lib/README.md]: remove as this is not required
its dynamic/automatic

File: projects/mathjax-lib/README.md [hunk 10]
```diff
@@ -1,24 +1,144 @@
...
+
+You can use the `MathJaxDirective.MathJaxTypeset()` method to trigger
+the typesetting when you want. The steps are:
+
+- Use *ViewChild* and its *read* property to get a reference to the
+  `MathJaxDirective` instance
+- Call `instance.MathJaxTypeset()`
+
+## Adding a callback
```
Comment (sajivkumar) [inline projects/mathjax-lib/README.md]: remove

File: package.json [hunk 11]
```diff
@@ -46,7 +46,7 @@
   "jest": {
```
Comment (sajivkumar) [inline package.json:46]: update 
`"version": "1.0.0",`
to
`"version": "1.1.0",`

File: projects/mathjax-lib/README.md (hunk 5 above)
Comment (codeaye) [reply projects/mathjax-lib/README.md:46]: https://cdnjs.com/libraries/mathjax

File: projects/mathjax-lib/README.md (hunk 7 above)
Comment (codeaye) [reply projects/mathjax-lib/README.md]: ## Typeset an element

Add the `mathjax` directive to elements which you want to apply
MathJax typesetting on load.

```html
<div [mathjax]="content"></div> 
// in your ts component content = "$x = {-b \\pm \\sqrt{b^2-4ac} \\over 2a}$"
```

File: projects/mathjax-lib/README.md [hunk 12]
```diff
@@ -1,24 +1,144 @@
-# MathjaxLib
+# mathjax-angular 
+This plugins implements the browser version of [Mathjax v3][1] into [Angular][2]
+## Feature
 
-This library was generated with [Angular CLI](https://github.com/angular/angular-cli) version 12.1.1.
+- Dynamically load MathJax library to your web application.
+- Simple typesetting using Angular directive.
```
Comment (sajivkumar) [inline projects/mathjax-lib/README.md]: remove

File: projects/mathjax-lib/README.md [hunk 13]
```diff
@@ -1,24 +1,144 @@
...
+- Dynamic typesetting using expressions.
```
Comment (sajivkumar) [inline projects/mathjax-lib/README.md]: remove
